javaparser-core/src/main/java/com/github/javaparser/printer/configuration/PrinterConfiguration.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (C) 2011, 2013-2020 The JavaParser Team.
+ *
+ * This file is part of JavaParser.
+ *
+ * JavaParser can be used either under the terms of
+ * a) the GNU Lesser General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ * b) the terms of the Apache License
+ *
+ * You should have received a copy of both licenses in LICENCE.LGPL and
+ * LICENCE.APACHE. Please refer to those files for details.
+ *
+ * JavaParser is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ */
+
+package com.github.javaparser.printer.configuration;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import com.github.javaparser.printer.Printable;
+import com.github.javaparser.printer.configuration.Indentation.IndentType;
+import com.github.javaparser.utils.Utils;
+
+/**
+ * Configuration options for the {@link Printable}.
+ */
+public class PrinterConfiguration implements ConfigurablePrinter {
+    
+    public enum ConfigOption {
+        ORDER_IMPORTS(Boolean.class), 
+        PRINT_COMMENTS(Boolean.class), 
```

Comment:
Does `PRINT_COMMENTS` imply `PRINT_JAVADOC`? Or vice-versa? What happens if one is true but the other is not?

I've not read the rest of it yet, maybe `PRINT_COMMENTS` refers to line comments and block comments? If so, let's make this explicit.

Reply:
It's from the legacy code see methods on PrettyPrinterConfiguration. I don't know exactly what they do.